string_grouper/string_grouper.py

```diff
@@ -8,13 +8,14 @@
 from sparse_dot_topn import awesome_cossim_topn
 from functools import wraps
 
-
 DEFAULT_NGRAM_SIZE: int = 3
 DEFAULT_REGEX: str = r'[,-./]|\s'
 DEFAULT_MAX_N_MATCHES: int = 20
 DEFAULT_MIN_SIMILARITY: float = 0.8  # Minimum cosine similarity for an item to be considered a match
 DEFAULT_N_PROCESSES: int = multiprocessing.cpu_count() - 1
 DEFAULT_IGNORE_CASE: bool = True  # ignores case by default
+DEFAULT_ID_COL: pd.Series = None  # No named column to be used for ID by default
```

Comment:
Could you change into:
DEFAULT_ID_COL: Optional[pd.Series] 

Otherwise my editor starts complaining :)

Reply:
:-) my editor complained also, thanks for the type hint.

Changed, will push with other changes below you've requested.

Just to share — this is a collaboration between myself and my friend who has mathematical & data science knowledge but less Python experience. We'll get through it.

I'm keen to see the package expanded in functionality - from personal interest obviously because I will have ongoing good use for it and because I want to promote it in the Python groups I'm in.